src/sensai/vector_model.py

```diff
@@ -379,13 +312,39 @@ def getModelOutputVariableNames(self):
     def getPredictedVariableNames(self):
```

Comment:
This can give incorrect results in the rule based models. Could be settled by setting `_predictedVariableNames` in the rule based models' constructors

Reply:
Same for `_modelOutputVariableNames `, which for rule based models should coincide (if the user did't come up with the idea of using a name-changing target transformer). I would say we don't need this variable, but we have the getter as part of the public interface

Reply:
Ah, maybe there is an acceptable way around: use the new `_underlyingModelRequiresFitting` flag to give a warning or even raise an exception in the getters. The docu should then say: not supported for rule based models.

Reply:
It will be set if users call `fit`, which could be appropriate, since there may be parts that still require fitting. 
We could solve it by extending the documentation of the method ("will return None if `fit` has not been called").

Reply:
I like your proposal. Make this getter log a warning if the value is None and the flag is true.
The other getter (`getModelOutputVariableNames`) should raise an exception.

Reply:
`_predictedVariableNames` won't always be set in fit for rule based models (see my comment for fit below) - Y can be None

Reply:
True. This dilutes the interface of `fit` somewhat, because, strictly speaking, we would need to annotate Y with `Optional[DataFrame]`. In Java, we wouldn't have this problem. :-/

Reply:
It just occurred to me that evaluation makes use of `getPredictedVariableNames` so we really need to get these names one way or another.